sstables/sstables_manager.cc

```diff
@@ -196,6 +196,15 @@ future<> sstables_manager::components_reloader_fiber() {
         while (!_reclaimed.empty() && memory_available > 0) {
             auto sstable_to_reload = _reclaimed.begin();
             const size_t reclaimed_memory = sstable_to_reload->total_memory_reclaimed();
+
+            if (sstable_to_reload->is_unlinked()) {
+                // This sstable has been unlinked already,
+                // remove it from reclaimed and try the next sstable.
+                _reclaimed.erase(sstable_to_reload);
+                _total_memory_reclaimed -= reclaimed_memory;
```

Comment:
Can we just add `_total_memory_reclaimed = 0` to `sstable::unlink()` so it is not even considered by this loop instead?

Reply:
@denesb realised after typing the above comment - I can just free the memory in unlink() and set the  `_total_memory_reclaimed` to 0. That will fix the issue.

Reply:
The patch sets the _total_memory_reclaimed on unlink now.